### PR TITLE
DAOS-5544 build: Add required dependencies to daos_test RPM

### PR DESCRIPTION
**Pull request by phender**
Separate the daos-tests RPM package into unique packages based upon
their external dependencies.

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

**Comment by phender** [comment]
Function on Leap 15 failure caused by https://jira.hpdd.intel.com/browse/DAOS-5781

File: ci/rpm/test_daos_node.sh [hunk 1]
```diff
@@ -4,7 +4,10 @@ set -uex
 sudo yum -y install --exclude ompi daos{,-client}-"${DAOS_PKG_VERSION}"
 sudo yum -y history rollback last-1
 sudo yum -y install --exclude ompi daos{,-{server,client}}-"${DAOS_PKG_VERSION}"
-sudo yum -y install --exclude ompi daos{,-tests}-"${DAOS_PKG_VERSION}"
+sudo yum -y install --exclude ompi daos-tests{,-common}-"${DAOS_PKG_VERSION}"
+sudo yum -y install --exclude ompi \
+  daos-tests-{ior,fio,mpiio,hdf5-vol,macsio,soak}-"${DAOS_PKG_VERSION}"
```
Comment (brianjmurrell) [inline ci/rpm/test_daos_node.sh:9]: These should all be install-able in the same `yum` command.

File: utils/check/handle_rpms.sh [hunk 2]
```diff
@@ -38,16 +38,53 @@ if ! grep "gpgcheck = false" "${repo_file}"; then
...
-                   libisa-l_crypto libopa1 libpmem libpmemblk mercury ndctl\
-                   ompi openmpi3 openpa-devel \
-                   pexpect protobuf-c pmix \
-                   python-configshell python-urwid \
-                   raft raft-debuginfo raft-devel \
-                   spdk spdk-tools
+sudo yum -y remove cart \
+                   cart-debuginfo \
+                   cart-devel \
+                   cart-tests \
```
Comment (brianjmurrell) [inline utils/check/handle_rpms.sh:44]: Cart is no longer in separate RPMs.  If you need to re-spin this PR, this seems like as good as any to remove these from here.

File: utils/rpms/daos.spec [hunk 3]
```diff
@@ -2,6 +2,36 @@
...
+      for ftest_file in ${ftests}; \
+      do \
+        for other_file in $(find ${ftest_file%%.*}.* | sort); \
+        do \
+          echo "${other_file#%3}" >> %1; \
+        done; \
+      done; \
+    done; \
+  done; \
+  cat %1 | sort | uniq > %{1}_unique; \
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:30]:
```suggestion
  cat %1 | sort -u > %{1}_unique; \
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Looks like a couple of test failures also.

File: ci/rpm/test_daos_node.sh (hunk 1 above)
Comment (phender) [reply ci/rpm/test_daos_node.sh:9]: There's a little history on this one.  I had tried to use one command with:
```
sudo yum -y install --exclude ompi \
  daos-tests{,-{common,ior,fio,mpiio,hdf5-vol,macsio,soak}}-\
  "${DAOS_PKG_VERSION}"
```
but the assembly of the string included the spaces and caused an error with the command.

I then tried this edit but it exceeded 80 chars:
```
sudo yum -y install --exclude ompi \
  daos-tests{,-{common,ior,fio,mpiio,hdf5-vol,macsio,soak}}-"${DAOS_PKG_VERSION}"
```

So the current implementation was the compromise, but if you have a better solution I'll give it a try.

File: utils/check/handle_rpms.sh (hunk 2 above)
Comment (phender) [reply utils/check/handle_rpms.sh:44]: Sounds good.

File: ci/rpm/test_daos_node.sh (hunk 1 above)
Comment (brianjmurrell) [reply ci/rpm/test_daos_node.sh:9]: I'd be fine with it being an 80-char limit exception.  But alternatively I think you can do:

```
sudo yum -y install --exclude ompi \
  daos-tests{,-{common,ior,fio,mpiio,hdf5-vol,macsio,soak}}-\
"${DAOS_PKG_VERSION}"
```

Albeit it does mess with the visual representation that the indent provides.

File: utils/rpms/daos.spec (hunk 3 above)
Comment (phender) [reply utils/rpms/daos.spec]: Done

**Comment by brianjmurrell** [comment]
Just a head's up, you have conflicts in Jenkinsfile and utils/rpms/daos.spec.  That latter one is a pain, to be sure.

**Comment by brianjmurrell** [comment]
Just a heads-up that you are racing with (at least) @mjean308 on `daos.spec` changes.  If you get to the point of fully passing and having your +1s, and end up having to push another merge commit just to update the `Release:` and `%changelog`, I have been successful in requesting a force-landing on such a PR just so that it doesn't waste resources testing just an RPM `Release:` update only to possibly lose a race again.

**Comment by phender** [comment]
@brianjmurrell do you have any insight into how to apply the approach of dynamically assigning the file list (`<package>.install` file) for a debian package.  It seems like using dedicated `ftest` sub-directories, e.g. `ior`, `fio`, `macsio`, `soak`, etc., would make it easier to define the same packages for both RPMs and debian packages.

**Comment by brianjmurrell** [comment]
> @brianjmurrell do you have any insight into how to apply the approach of dynamically assigning the file list (`<package>.install` file) for a debian package. It seems like using dedicated `ftest` sub-directories, e.g. `ior`, `fio`, `macsio`, `soak`, etc., would make it easier to define the same packages for both RPMs and debian packages.

Not in particular.  You could try to edit/generate the `*.install` files in https://github.com/daos-stack/daos/blob/29fae9109800b09490eea10cb560f853fe634da7/debian/rules#L47.  Let me know if/how that works out.

**Comment by phender** [comment]
Test failures in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3998/4/:
- Functional_Hardware_Large / PoolCreateTests.3-./pool/create_test.py:  https://jira.hpdd.intel.com/browse/DAOS-6200

**Pull request closed without merging**